### PR TITLE
sndfile Module - Add Custom snd_name in ~/.baresip/config

### PR DESCRIPTION
**Pull request by ninp0**
`snd_name` in ~/.baresip/config can be:
* `curi` (local_uri)
* `duri` (peer_uri)
* or `both` 

This is mostly working w/ the exception of one issue:

```
baresip $ make
  CC [M]  build-x86_64/modules/sndfile/sndfile.o
modules/sndfile/sndfile.c: In function ‘openfile’:
modules/sndfile/sndfile.c:98:36: error: invalid use of undefined type ‘struct stream’
   98 |                 snd_name_opt = strm->cname;
      |                                    ^~
modules/sndfile/sndfile.c:101:36: error: invalid use of undefined type ‘struct stream’
  101 |                 snd_name_opt = strm->duri;
      |                                    ^~
modules/sndfile/sndfile.c:106:39: error: invalid use of undefined type ‘struct stream’
  106 |                                   strm->cname,
      |                                       ^~
modules/sndfile/sndfile.c:107:39: error: invalid use of undefined type ‘struct stream’
  107 |                                   strm->duri);
      |                                       ^~
make: *** [mk/mod.mk:49: build-x86_64/modules/sndfile/sndfile.o] Error 1
```

`modules/sndfile/sndfile.c` includes `baresip.h` which references `struct stream`
I'm also passing the `*strm` pointer as `struct stream` to the `openfile` function and instantiating the `*strm` pointer via:
```
struct stream *strm = audio_strm(au);
```
in both the `encode_update` and `decode_update` functions before `(void)au` is called.
what am I missing?

### CODE REVIEW

File: include/baresip.h [hunk 1]
```diff
@@ -1368,6 +1368,7 @@ struct stream_param {
 	bool use_rtp;       /**< Enable or disable RTP */
 	int af;             /**< Wanted address family */
 	const char *cname;  /**< Canonical name        */
+	const char *duri;
```
Comment (cspiel1) [inline include/baresip.h:1371]: `duri` or `puri` or simply `peer`? And pls add a Doxygen!

```c
const char *peer;   /**< Peer uri/name or identifier  */
```

**Comment by cspiel1** [comment]
There is one commit that is confusing. The "Merge branch" commit. You should use `git rebase -i main`, interactive rebase to rebase your changes onto main HEAD.
```
git checkout main
git pull
git checkout sndfile_peer_uri_in_filename
git rebase -i main
```

**Comment by sreimers** [comment]
> `modules/sndfile/sndfile.c` includes `baresip.h` which references `struct stream`
> I'm also passing the `*strm` pointer as `struct stream` to the `openfile` function and instantiating the `*strm` pointer via:

`struct stream` is private (only file scope) and only forwared in `baresip.h`. So you can't access members directly.
You need to add a getter function in `stream.c`

File: src/stream.c [hunk 2]
```diff
@@ -50,36 +50,37 @@ struct receiver {
...
-	bool menc_secure;        /**< Media stream is secure                */
-	stream_pt_h *pth;        /**< Stream payload type handler           */
-	stream_rtp_h *rtph;      /**< Stream RTP handler                    */
-	stream_rtcp_h *rtcph;    /**< Stream RTCP handler                   */
-	void *arg;               /**< Handler argument                      */
-	stream_mnatconn_h *mnatconnh;/**< Medianat connected handler        */
-	stream_rtpestab_h *rtpestabh;/**< RTP established handler           */
-	stream_rtcp_h *sessrtcph;    /**< Stream RTCP handler               */
-	stream_error_h *errorh;  /**< Stream error handler                  */
-	void *sess_arg;          /**< Session handlers argument             */
```
Comment (cspiel1) [inline src/stream.c:82]: A lot of unrelated changes. Pls avoid this!

File: src/stream.c [hunk 3]
```diff
@@ -50,36 +50,37 @@ struct receiver {
...
+	struct rtp_sock *rtp;    /**< RTP Socket                           */
+	struct rtcp_stats rtcp_stats;/**< RTCP statistics                  */
+	const struct mnat *mnat; /**< Media NAT traversal module           */
+	struct mnat_media *mns;  /**< Media NAT traversal state            */
+	const struct menc *menc; /**< Media encryption module              */
+	struct menc_sess *mencs; /**< Media encryption session state       */
+	struct menc_media *mes;  /**< Media Encryption media state         */
+	enum media_type type;    /**< Media type, e.g. audio/video         */
+	char *cname;             /**< RTCP Canonical incoming end-point    */
+	char *duri;              /**< RTCP Canonical outgoing end-point    */
```
Comment (cspiel1) [inline src/stream.c:68]: I think this has nothing to do with `RTCP`. Better:

```c
char *peer;              /**< Peer name or identifier      */
```

File: src/call.c [hunk 4]
```diff
@@ -795,6 +795,7 @@ int call_streams_alloc(struct call *call)
 	strm_prm.use_rtp = call->use_rtp;
 	strm_prm.af      = call->af;
 	strm_prm.cname   = call->local_uri;
+	strm_prm.duri   = call->peer_uri;
```
Comment (cspiel1) [inline src/call.c:798]: pls align the "="

File: src/config.c [hunk 5]
```diff
@@ -1184,6 +1184,7 @@ int config_write_template(const char *file, const struct config *cfg)
 
 	(void)re_fprintf(f,
 			 "\n# sndfile\n"
+			 "#snd_name\t\tcuri, duri, OR both"
```
Comment (cspiel1) [inline src/config.c:1187]: Do we need a new config?

File: src/stream.c [hunk 6]
```diff
@@ -50,36 +50,37 @@ struct receiver {
...
+	bool menc_secure;        /**< Media stream is secure               */
+	stream_pt_h *pth;        /**< Stream payload type handler          */
+	stream_rtp_h *rtph;      /**< Stream RTP handler                   */
+	stream_rtcp_h *rtcph;    /**< Stream RTCP handler                  */
+	void *arg;               /**< Handler argument                     */
+	stream_mnatconn_h *mnatconnh;/**< Medianat connected handler       */
+	stream_rtpestab_h *rtpestabh;/**< RTP established handler          */
+	stream_rtcp_h *sessrtcph;    /**< Stream RTCP handler              */
+	stream_error_h *errorh;  /**< Stream error handler                 */
+	void *sess_arg;          /**< Session handlers argument            */
```
Comment (cspiel1) [inline src/stream.c:83]: A lot of unrelated changes. Pls avoid this!

File: modules/sndfile/sndfile.c [hunk 7]
```diff
@@ -88,9 +92,25 @@ static SNDFILE *openfile(const struct aufilt_prm *prm, bool enc)
...
+	if (0 == strcmp(snd_name, "curi")) {
+		snd_name_opt = strm->cname;
+	}
+	else if (0 == strcmp(snd_name, "duri")) {
+		snd_name_opt = strm->duri;
+	}
+	else {
+		(void)re_snprintf(snd_name, sizeof(snd_name),
+				  "%s<->%s",
+				  strm->cname,
```
Comment (cspiel1) [inline modules/sndfile/sndfile.c:106]: Can we simplify this? Would it be ok if we have only one of this three variants and avoid the new config setting. Maybe this would be enough?

```c
	(void)re_snprintf(filename, sizeof(filename),
			  "%s/dump-%s<->%s-%H-%s.wav",
			  stream_cname(strm), stream_peer(strm),
			  timestamp_print, tm, enc ? "enc" : "dec");
```

You see also: You have to use getter functions for `struct stream`. This have to be added to the API (baresip.h).

File: modules/sndfile/sndfile.c [hunk 8]
```diff
@@ -79,7 +81,9 @@ static int get_format(enum aufmt fmt)
 }
 
 
-static SNDFILE *openfile(const struct aufilt_prm *prm, bool enc)
+static SNDFILE *openfile(const struct aufilt_prm *prm,
+			 struct stream *strm,
+			 bool enc)
```
Comment (cspiel1) [inline modules/sndfile/sndfile.c:86]: Just a minor comment: You may put stream in first line. We have 79 chars per line.
```c
static SNDFILE *openfile(const struct aufilt_prm *prm, struct stream *strm,
			 bool enc)
```

**Comment by ninp0** [comment]
Thank you so much for you quick turn-around and feedback...I'll PR the changes based on your feedback shortly.
